content/posts/jenkins-x-flow-integration.md

```diff
@@ -0,0 +1,147 @@
+date: 2019-05-28T10:58:46-04:00
+showDate: true
+draft: true
+tags: ["jenkins","jenkins x","cloudbees flow", "tekon", "extending jenkins x"]
+---
+[Jenkins X](https://jenkins-x.io) is quickly becoming the de facto standard for high performing teams wanting to do CI/CD in a highly scalable and fault tolerant environment. For those who haven’t gotten the opportunity to try out Jenkins X, it allows teams to run CI/CD workloads natively in a Kubernetes environment while taking advantage of modern operating patterns like GitOps and serverless architectures. For teams wanting to modernize their continuous integration and continuous deployment capabilities, Jenkins X is the go to solution.
+
+In today’s heterogenous technology environment, most organizations tend to have a mix of modern cloud native architectures as well as more traditional workloads which get deployed either on-prem or within the cloud. In the latter case, a combination of Jenkins X (performing CI steps) and CloudBees Flow (handling the deployment) can add a huge amount of flexibility and power to a Continuous Delivery process.  Jenkins X can be easily extended to accommodate any type of workload required - it can be a full end to end CI/CD tool for building, deploying, and running applications all within a Kubernetes cluster, or it can handle CI while offloading other release and deployment tasks to another solution.  In this blog post we’re going to cover how Jenkins X can be extended to offload release/deployment tasks to [CloudBees Flow](https://www.cloudbees.com/cloudbees-acquires-electric-cloud).
+
+# Extending Jenkins X
+For the purposes of this blog, we’re going to be focusing on the Jenkins X serverless pipeline execution engine with Tekton (See https://jenkins-x.io/architecture/jenkins-x-pipelines/). There are two main ways to customize a Jenkins X pipeline in order to integrate with CloudBees Flow.  The first and simplest would be to modify the jenkins-x.yml (more information on Jenkins X pipelines: https://jenkins-x.io/architecture/jenkins-x-pipelines/#differences-to-jenkins-pipelines and the jenkins-x.yml file) pipeline file in the source code repo for the project we’re going to build.  The other way is to extend the [Jenkins X build packs](https://jenkins-x.io/architecture/build-packs/) and modify the build pack for the language/build tool you want to use.  Both will work, but by forking the build packs you can get reuse across multiple projects using the build pack you extend. In this example, we’ll walk through how to extend the Jenkins X build packs.
+
+## Creating our Cluster and Installing Jenkins X
+To start, we’ll fork the Jenkins X Kubernetes build packs into our own repository: https://github.com/jenkins-x-buildpacks/jenkins-x-kubernetes.  Later we'll be extending the maven build pack to support a REST API call into CloudBees Flow.
+
+Then it’s time to create a Kubernetes cluster on GKE using Jenkins X and [Tekton](https://github.com/tektoncd/pipeline): 
```

Comment:
Maybe should mention why you are creating the cluster. Or just mentioning the option of installing Jenkins X in your own cluster if you already have it and have some limitations.

# Installing Jenkins X in your own GKE cluster
If you want to install Jenkins X in your existint GKE cluster you can do it by the `jx install`command. In this case your cluster must have the following "scopes" enabled:
- monitoring
- logging-write
- cloud-platform
- service-management
- compute-rw
- service-control
- storage-full
```
So you can install with the following command (change parameters with your values):
jx install --provider gke \
--default-admin-password $MY_PWD \
--default-environment-prefix $JX_CLUSTER \
--environment-git-owner $GH_ORG \
--docker-registry=gcr.io \
--docker-registry-org=${GCP_PROJECT} \
--kaniko \
--tekton --no-tiller
```